### PR TITLE
post a project process

### PR DESCRIPTION
**Pull request by sultanassi95**
#34 #43
what happend:

1. inserted a user (gwo), with id (1) into users table.
2. inserted a category (Web Development), with id (1) into category table.
3. added file: src/database/queries/insertProject.js
3. a: wrote query into the file to perform inserting a project into projects table into the database.
4. added route ('/post-project') into (mainController) that handles inserting project requests.
4. a: invoked insertProject query and handled it's results.
5. Tested the applied route with success, response on route:
```
{
result: {
command: "INSERT",
rowCount: 1,
oid: 0,
rows: [ ],
fields: [ ],
_parsers: [ ],
RowCtor: null,
rowAsArray: false
},
responseText: "Inserted Project Successfully"
}
```

### CODE REVIEW

**Review by NouraldinS — APPROVED** [review]
I'd go more with naming the function addProject rather than insertProject, perhaps more comments defining the hardcode part. But generally, great job.

**Review by yahyaHB — APPROVED** [review]
well done mate, be more descriptive with the naming from now on

File: src/controllers/mainController.js [hunk 1]
```diff
@@ -13,4 +14,32 @@ router.get('/', (req, res) => {
   // return res.send('Hello world');
```
Comment (mohanagy) [inline src/controllers/mainController.js]: remove this line

File: src/database/queries/insertProject.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+const dbConnection = require('../dbConnection');
+
+const insertProject = (projectObj, cb) => {
+  const insertProjectQuery = {
+    text: 'insert into projects (title, user_id, category_id, skills, description, budget, dead_line, lifetime) values ($1, $2, $3, $4, $5, $6, $7, $8)',
```
Comment (mohanagy) [inline src/database/queries/insertProject.js]: max line length shouldn't be more than 100

use   ` instead of '

**Review by mohanagy — CHANGES_REQUESTED** [review]

**Pull request closed without merging**